### PR TITLE
Add user authentication feature

### PR DESCRIPTION
**Pull request by lhlopezm2**
Add user authentication feature and apply rubocop guidelines

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 # Controller for articles model
 class ArticlesController < ApplicationController
   # http_basic_authenticate_with name: "dhh", password: "secret", except: [:index, :show]
+  # before_action :authenticate_user!, except: [:index, :show]
```
Comment (MrJuanGaviriaK) [inline app/controllers/articles_controller.rb:6]: I think we need this line, to protect some actions for the article controller(for example show), what do you think?

File: app/views/articles/index.html.erb [hunk 2]
```diff
@@ -12,4 +12,6 @@ Our blog has <%= Article.public_count %> articles and counting!
   <% end %>
 </ul>
 
-<%= link_to "New Article", new_article_path %>
+<% if signed_in? %>
+  <%= link_to "New Article", new_article_path %>
```
Comment (MrJuanGaviriaK) [inline app/views/articles/index.html.erb:16]: You should use simple quotes.

File: app/views/articles/show.html.erb [hunk 3]
```diff
@@ -2,16 +2,20 @@
 
 <p><%= @article.body %></p>
 
-<ul>
-  <li><%= link_to "Edit", edit_article_path(@article) %></li>
-  <li><%= link_to "Destroy", article_path(@article), data: {
+<% if signed_in? %>
+  <ul>
+    <li><%= link_to "Edit", edit_article_path(@article) %></li>
```
Comment (MrJuanGaviriaK) [inline app/views/articles/show.html.erb:7]: Idem.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -11,6 +11,14 @@
   </head>
 
   <body>
+    <div>
+      <% if signed_in? %>
+        <%= link_to "Log out", destroy_user_session_path, method: :delete %>
```
Comment (MrJuanGaviriaK) [inline app/views/layouts/application.html.erb:16]: Idem.

File: test/fixtures/users.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (MrJuanGaviriaK) [inline test/fixtures/users.yml:1]: Do you need this file?

File: test/models/user_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (MrJuanGaviriaK) [inline test/models/user_test.rb:1]: Idem

**Review by MrJuanGaviriaK — CHANGES_REQUESTED** [review]
Hi, I left some comments.

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (lhlopezm2) [reply app/controllers/articles_controller.rb:6]: I think it depends on the purpose. There, I just hide the add, comment and delete buttons for non-logged users.

File: test/fixtures/users.yml (hunk 5 above)
Comment (lhlopezm2) [reply test/fixtures/users.yml:1]: This file comes predefined with the project. I think we could use it later.

**Review by MrJuanGaviriaK — APPROVED** [review]

**Pull request merged**